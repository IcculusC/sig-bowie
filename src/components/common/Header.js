import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import IconButton from 'material-ui/IconButton';
import SvgIcon from 'material-ui/SvgIcon';


const styles = {
  root: {
    flexGrow: 1,
  },
};

const propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
  }).isRequired,
};

const BlackstarIcon = props => (
  <SvgIcon {...props} >
    <path d='m55,237 74-228 74,228L9,96h240'/>
  </SvgIcon>
)

export const HeaderComponent = ({ classes }) => (
  <div className={classes.root} >
    <AppBar position="static" color="default">
      <Toolbar>
        <IconButton className={classes.blackstar} color="inherit" aria-label="Bowie Baby">
          <BlackstarIcon viewBox='0 0 260 260' nativeColor='#000' />
        </IconButton>
        <Typography variant="title" color="inherit">
            Bowie Baby
        </Typography>
      </Toolbar>
    </AppBar>
  </div>
);

HeaderComponent.propTypes = propTypes;
HeaderComponent.displayName = 'Header';

export const Header = withStyles(styles)(HeaderComponent);
