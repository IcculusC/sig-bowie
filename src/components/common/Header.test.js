import React from 'react';
import renderer from 'react-test-renderer';
import { BlackstarIcon, Header } from './Header';


describe('Header component', () => {
  it('renders properly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Blackstar icon properly', () => {
    const tree = renderer.create(<BlackstarIcon  viewBox='0 0 260 260' nativeColor='#000' />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
