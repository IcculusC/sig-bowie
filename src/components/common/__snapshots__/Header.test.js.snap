// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Blackstar icon properly 1`] = `
<svg
  aria-hidden="true"
  className="MuiSvgIcon-root-47"
  color="#000"
  focusable="false"
  viewBox="0 0 260 260"
>
  <path
    d="m55,237 74-228 74,228L9,96h240"
  />
</svg>
`;

exports[`Header component renders properly 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorDefault-7"
  >
    <div
      className="MuiToolbar-root-37 MuiToolbar-gutters-38"
    >
      <button
        aria-label="Bowie Baby"
        className="MuiButtonBase-root-45 MuiIconButton-root-39 MuiIconButton-colorInherit-40"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-44"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-47"
            color="#000"
            focusable="false"
            viewBox="0 0 260 260"
          >
            <path
              d="m55,237 74-228 74,228L9,96h240"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-53"
        />
      </button>
      <h2
        className="MuiTypography-root-60 MuiTypography-title-66 MuiTypography-colorInherit-79"
      >
        Bowie Baby
      </h2>
    </div>
  </header>
</div>
`;
